### PR TITLE
Selenium: Cover "Dashboard / Workspace Details / Overview" part of test plan by selenium test

### PR DESCRIPTION
**Pull request by Ohrimenko1988**
### What does this PR do?
Cover "Dashboard / Workspace Details / Overview" part of test plan by selenium test

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/10631

### CODE REVIEW

File: selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/dashboard/AddOrImportForm.java [hunk 1]
```diff
@@ -126,6 +126,19 @@ public void clickOnAddButton() {
     seleniumWebDriverHelper.waitAndClick(By.id(ADD_BUTTON_ID));
   }
 
+  public void addSampleToWorkspace(String sampleName) {
+    waitAddOrImportFormOpened();
+
+    if (!isSampleCheckboxEnabled(sampleName)) ;
+    {
```
Comment (dmytro-ndp) [inline selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/dashboard/AddOrImportForm.java]: `;` is redundant here

File: selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/dashboard/AddOrImportForm.java [hunk 2]
```diff
@@ -299,6 +312,14 @@ public void clickOnSampleCheckbox(String sampleName) {
         By.xpath(format("//*[@id='%s']/md-checkbox/div", checkboxId)));
   }
 
+  public boolean isSampleCheckboxEnabled(String sampleName) {
+    String checkboxId = format(CHECKBOX_BY_SAMPLE_NAME_ID_TEMPLATE, sampleName);
+    String locator = format("//div[@id='%s']/md-checkbox", checkboxId);
```
Comment (dmytro-ndp) [inline selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/dashboard/AddOrImportForm.java]: Locator of what is this?

File: selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/dashboard/workspaces/WorkspaceOverview.java [hunk 3]
```diff
@@ -50,6 +65,13 @@ public WorkspaceOverview(SeleniumWebDriver seleniumWebDriver) {
     String HIDE_JSON_WS_BTN = "//span[text()='Close']";
     String WORKSPACE_JSON_CONTENT = "//div[@class='CodeMirror-code']";
     String WS_NAME_ERROR_MESSAGES = "//che-error-messages";
+    String EXPORT_WS_FORM_XPATH = "//che-popup[@title='Export Workspace']";
+    String AS_FILE_CONFIG_BODY_XPATH =
+        "//che-popup[@title='Export Workspace']//div[@class='CodeMirror-code']";
+
+    String SAVE_BUTTON_XPATH = "//button[@name='save-button' ]";
+    String AS_FILE_BUTTON_XPATH = "//md-tab-item/span[text()='As a File']";
+    String To_PRIVATE_CLOUD_BUTTON_XPATH = "//md-tab-item/span[text()='To Private Cloud']";
```
Comment (dmytro-ndp) [inline selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/dashboard/workspaces/WorkspaceOverview.java]: `To_` looks redundant in the constant name `To_PRIVATE_CLOUD_BUTTON_XPATH`.

File: selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/dashboard/workspaces/WorkspaceOverview.java [hunk 4]
```diff
@@ -87,6 +109,119 @@ public void checkNameWorkspace(String nameWorkspace) {
...
 
+  public void waitNameFieldValue(String workspaceName) {
+    seleniumWebDriverHelper.waitSuccessCondition(
+        driver ->
+            seleniumWebDriverHelper
+                .waitVisibilityAndGetValue(nameWorkspaceInput)
+                .equals(workspaceName));
+  }
+
+  public void waitUntillNoErrorsDisplayed() {
```
Comment (dmytro-ndp) [inline selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/dashboard/workspaces/WorkspaceOverview.java]: `Untill` > `Until`

File: selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/dashboard/workspaces/WorkspaceOverview.java [hunk 5]
```diff
@@ -87,6 +109,119 @@ public void checkNameWorkspace(String nameWorkspace) {
...
+  }
+
+  public void waitExportWorkspaceFormClosed() {
+    seleniumWebDriverHelper.waitInvisibility(By.xpath(EXPORT_WS_FORM_XPATH));
+  }
+
+  public boolean isAsFileTabOpened() {
+    return seleniumWebDriverHelper
+        .waitVisibilityAndGetAttribute(
+            By.xpath(AS_FILE_BUTTON_XPATH + "/parent::md-tab-item"), "aria-selected")
```
Comment (dmytro-ndp) [inline selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/dashboard/workspaces/WorkspaceOverview.java]: It's not quite clear which element xpath `AS_FILE_BUTTON_XPATH + "/parent::md-tab-item"` is linked to.

File: selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/dashboard/workspaces/WorkspaceOverview.java [hunk 6]
```diff
@@ -87,6 +109,119 @@ public void checkNameWorkspace(String nameWorkspace) {
...
+    seleniumWebDriverHelper.waitSuccessCondition(driver -> !isAsFileTabOpened());
+  }
+
+  public void clickOnToPrivateCloudButton() {
+    seleniumWebDriverHelper.waitAndClick(By.xpath(To_PRIVATE_CLOUD_BUTTON_XPATH));
+  }
+
+  public boolean isToPrivateCloudTabOpened() {
+    return seleniumWebDriverHelper
+        .waitVisibilityAndGetAttribute(
```
Comment (dmytro-ndp) [inline selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/dashboard/workspaces/WorkspaceOverview.java]: method `seleniumWebDriverHelper::waitAttributeEqualsTo()` is more suitable here, IMHO

File: selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/dashboard/workspaces/WorkspaceOverview.java [hunk 7]
```diff
@@ -87,6 +109,119 @@ public void checkNameWorkspace(String nameWorkspace) {
...
+            By.xpath(To_PRIVATE_CLOUD_BUTTON_XPATH + "/parent::md-tab-item"), "aria-selected")
```
Comment (dmytro-ndp) [inline selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/dashboard/workspaces/WorkspaceOverview.java]: It's not quite clear which element xpath `To_PRIVATE_CLOUD_BUTTON_XPATH + "/parent::md-tab-item"` is linked to.

File: selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/dashboard/workspaces/WorkspaceOverview.java [hunk 8]
```diff
@@ -87,6 +109,119 @@ public void checkNameWorkspace(String nameWorkspace) {
...
+    seleniumWebDriverHelper.waitAndClick(By.xpath(clipboardButtonXpath));
+  }
+
+  public void clickOnCloseExportWorkspaceButton() {
+    final String closeButtonXpath = "//che-button-notice [@che-button-title='Close']";
+    seleniumWebDriverHelper.waitAndClick(By.xpath(closeButtonXpath));
+  }
+
+  public void clickOnCloseExportWorkspaceIcon() {
+    seleniumWebDriverHelper.waitAndClick(By.xpath(EXPORT_WS_FORM_XPATH + "//i"));
```
Comment (dmytro-ndp) [inline selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/dashboard/workspaces/WorkspaceOverview.java]: It's not quite clear which element xpath `EXPORT_WS_FORM_XPATH + "//i"` is linked to.

File: selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/dashboard/workspaces/WorkspaceOverview.java [hunk 9]
```diff
@@ -87,6 +109,119 @@ public void checkNameWorkspace(String nameWorkspace) {
...
+    testWebElementRenderChecker.waitElementIsRendered(By.xpath(EXPORT_WS_FORM_XPATH));
+  }
+
+  public void waitExportWorkspaceFormClosed() {
+    seleniumWebDriverHelper.waitInvisibility(By.xpath(EXPORT_WS_FORM_XPATH));
+  }
+
+  public boolean isAsFileTabOpened() {
+    return seleniumWebDriverHelper
+        .waitVisibilityAndGetAttribute(
```
[the remark this answers is not in the document]
Comment (Ohrimenko1988) [reply selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/dashboard/workspaces/WorkspaceOverview.java]: In this case, we need the "boolean" returning type, in the case of seleniumWebDriverHelper::waitAttributeEqualsTo() method we have the "void" returning type

File: selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/dashboard/workspaces/WorkspaceOverview.java (hunk 6 above)
Comment (Ohrimenko1988) [reply selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/dashboard/workspaces/WorkspaceOverview.java]: https://github.com/eclipse/che/pull/10661#discussion_r207565445

**Comment by dmytro-ndp** [comment]
Do not forget to add new tests into the CheSuite.xml.

File: selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/dashboard/workspaces/details/WorkspaceDetailsOverviewTest.java [hunk 10]
```diff
@@ -0,0 +1,238 @@
...
+  private static final String MIN_SHORT_NAME = "wks";
+  private static final String SHORT_NAME_ERROR_MESSAGE =
+      "The name has to be more than 3 characters long.";
+  private static final String SPECIAL_CHARACTERS_ERROR_MESSAGE =
+      "The name should not contain special characters like space, dollar, etc.";
+  private static final List<String> VALID_NAMES =
+      asList("Wk-sp", "Wk-sp1", "9wk-sp", "5wk-sp0", "Wk19sp", "Wksp-01");
+  private static final List<String> NOT_VALID_NAMES =
+      asList("wksp-", "-wksp", "wk sp", "wk_sp", "wksp@", "wksp$", "wksp&", "wksp*");
+  private static final String EXPECTED_CONFIG =
```
Comment (SkorikSergey) [inline selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/dashboard/workspaces/details/WorkspaceDetailsOverviewTest.java]: Are you sure this config will be the same for openshift and docker?

File: selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/dashboard/workspaces/details/WorkspaceDetailsOverviewTest.java [hunk 11]
```diff
@@ -0,0 +1,238 @@
...
+    workspaceOverview.waitExportWorkspaceFormClosed();
+  }
+
+  private void selectStackAndCheckWorkspaceName(NewWorkspace.Stack stack) {
+    newWorkspace.selectStack(stack);
+    newWorkspace.waitStackSelected(stack);
+    newWorkspace.waitWorkspaceNameFieldValue(WORKSPACE_NAME);
+  }
+
+  private void setupValidName(String name) {
```
Comment (dmytro-ndp) [inline selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/dashboard/workspaces/details/WorkspaceDetailsOverviewTest.java]: `setupValidName` > `nameShouldBeValid(String name)`

File: selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/dashboard/workspaces/details/WorkspaceDetailsOverviewTest.java [hunk 12]
```diff
@@ -0,0 +1,238 @@
...
+    } catch (TimeoutException ex) {
+      // remove try-catch block after issue has been resolved
+      fail("Known issue https://github.com/eclipse/che/issues/10659", ex);
+    }
+
+    workspaceOverview.waitExpectedNameErrorMessage(expectedErrorMessage);
+    workspaceOverview.waitDisabledSaveButton();
+  }
+
+  private void checkValidNames() {
```
Comment (dmytro-ndp) [inline selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/dashboard/workspaces/details/WorkspaceDetailsOverviewTest.java]: `checkValidNames` > `namesShouldBeValid()`

File: selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/dashboard/workspaces/details/WorkspaceDetailsOverviewTest.java [hunk 13]
```diff
@@ -0,0 +1,238 @@
...
+    newWorkspace.waitWorkspaceNameFieldValue(WORKSPACE_NAME);
+  }
+
+  private void setupValidName(String name) {
+    workspaceOverview.enterNameWorkspace(name);
+    workspaceOverview.waitUntilNoErrorsDisplayed();
+    workspaceOverview.waitEnabledSaveButton();
+  }
+
+  private void setupInvalidNameAndCheckErrorMessage(String name, String expectedErrorMessage) {
```
Comment (dmytro-ndp) [inline selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/dashboard/workspaces/details/WorkspaceDetailsOverviewTest.java]: `setupInvalidNameAndCheckErrorMessage(String name)` > `nameShouldBeInvalid(String name, String expectedErrorMessage)`

File: selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/dashboard/workspaces/details/WorkspaceDetailsOverviewTest.java [hunk 14]
```diff
@@ -0,0 +1,238 @@
...
+
+  private void checkValidNames() {
+    VALID_NAMES.forEach(
+        name -> {
+          setupInvalidNameAndCheckErrorMessage(TOO_SHORT_NAME, SHORT_NAME_ERROR_MESSAGE);
+          setupValidName(name);
+        });
+  }
+
+  private void checkInvalidNames() {
```
Comment (dmytro-ndp) [inline selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/dashboard/workspaces/details/WorkspaceDetailsOverviewTest.java]: `checkInvalidNames()` > `namesShouldBeInvalid()`

File: selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/dashboard/workspaces/details/WorkspaceDetailsOverviewTest.java [hunk 15]
```diff
@@ -0,0 +1,238 @@
...
+    NOT_VALID_NAMES.forEach(
+        name -> {
+          setupValidName(CHANGED_WORKSPACE_NAME);
+          setupInvalidNameAndCheckErrorMessage(name, SPECIAL_CHARACTERS_ERROR_MESSAGE);
+        });
+  }
+
+  private void openExportWorkspaceForm() {
```
Comment (dmytro-ndp) [inline selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/dashboard/workspaces/details/WorkspaceDetailsOverviewTest.java]: `openExportWorkspaceForm()` could be useful for other tests and so more suitable for `WorkspaceOverview` page object.

File: selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/dashboard/workspaces/WorkspaceOverview.java [hunk 16]
```diff
@@ -87,6 +109,121 @@ public void checkNameWorkspace(String nameWorkspace) {
...
+    final String closeButtonXpath = "//che-button-notice [@che-button-title='Close']";
+    seleniumWebDriverHelper.waitAndClick(By.xpath(closeButtonXpath));
+  }
+
+  public void clickOnCloseExportWorkspaceIcon() {
+    final String closeFormIcon = EXPORT_WS_FORM_XPATH + "//i";
+    seleniumWebDriverHelper.waitAndClick(By.xpath(closeFormIcon));
+  }
+
+  public void waitExpectedNameErrorMessage(String expectedMessage) {
```
Comment (dmytro-ndp) [inline selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/dashboard/workspaces/WorkspaceOverview.java]: `waitExpectedNameErrorMessage()` > `waitNameErrorMessage`

File: selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/dashboard/workspaces/WorkspaceOverview.java [hunk 17]
```diff
@@ -87,6 +109,121 @@ public void checkNameWorkspace(String nameWorkspace) {
...
+    final String downloadButtonXpath = "//che-button-default [@che-button-title='download']";
+    seleniumWebDriverHelper.waitAndClick(By.xpath(downloadButtonXpath));
+  }
+
+  public void clickOnClipboardButton() {
+    final String clipboardButtonXpath = "//che-button-default [@che-button-title='clipboard']";
+    seleniumWebDriverHelper.waitAndClick(By.xpath(clipboardButtonXpath));
+  }
+
+  public void clickOnCloseExportWorkspaceButton() {
```
Comment (dmytro-ndp) [inline selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/dashboard/workspaces/WorkspaceOverview.java]: `clickOnCloseExportWorkspaceButton` > `clickOnCloseExportWorkspaceFormButton`

File: selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/dashboard/workspaces/WorkspaceOverview.java [hunk 18]
```diff
@@ -87,6 +109,121 @@ public void checkNameWorkspace(String nameWorkspace) {
...
+    final String closeButtonXpath = "//che-button-notice [@che-button-title='Close']";
+    seleniumWebDriverHelper.waitAndClick(By.xpath(closeButtonXpath));
+  }
+
+  public void clickOnCloseExportWorkspaceIcon() {
```
Comment (dmytro-ndp) [inline selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/dashboard/workspaces/WorkspaceOverview.java]: `clickOnCloseExportWorkspaceIcon` > `clickOnCloseExportWorkspaceFormIcon`

**Comment by dmytro-ndp** [comment]
It may be reasonable to create separate page object class `ExportWorkspaceForm` to make `WorkspaceDetails` page object class simpler.

File: selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/dashboard/workspaces/WorkspaceOverview.java (hunk 9 above)
[the remark this answers is not in the document]
Comment (dmytro-ndp) [reply selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/dashboard/workspaces/WorkspaceOverview.java]: You are right.

File: selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/dashboard/workspaces/WorkspaceOverview.java (hunk 6 above)
Comment (dmytro-ndp) [reply selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/dashboard/workspaces/WorkspaceOverview.java]: I see.

**Review by dmytro-ndp — APPROVED** [review]

**Review by SkorikSergey — APPROVED** [review]

**Review by musienko-maxim — APPROVED** [review]

**Pull request merged**